### PR TITLE
[4.0] Fix warning for HTMLHelper::_('calendar')

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue #34948.

### Summary of Changes
Fix warning for HTMLHelper::_('calendar')

### Testing Instructions
1. See https://github.com/joomla/joomla-cms/issues/34948 , confirm the issue
2. Apply patch, confirm the issue fixed.

### CODE REVIEW

**Comment by Fedik** [comment]
You was faster than me :)
Can you please add to the calendar layout extra check, if someoen use layout in own field:

```
'data-firstday'        => $firstday,
'data-weekend'         => implode(',', $weekend),
```
to
```
'data-firstday'        => empty($firstday) ? '' : $firstday,
'data-weekend'         => empty($weekend) ? '' : implode(',', $weekend),
```

**Comment by joomdonation** [comment]
@Fedik A small concern here. You removed the localesPath from layout data, so if someone overrides the joomla.form.field.calendar before, there would be warning. Not sure if that's something we have to care?

**Comment by Fedik** [comment]
yeah, unfortunately they have to sync the override then,
because there no any localesPath anymore, and providing an empty string just hide a notice, however we can do that

**Comment by joomdonation** [comment]
Ah, I see. It is because these js files were removed. OK, I committed the changes you requested. Should be fine now.

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 52e1ac57d4c5f32b301e30f6497d68229f79a38f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34949">issues.joomla.org/tracker/joomla-cms/34949</a>.</sub>

File: layouts/joomla/form/field/calendar.php [hunk 1]
```diff
@@ -103,8 +103,8 @@
 	'data-inputfield'      => $id,
 	'data-button'          => $id . '_btn',
 	'data-date-format'     => $format,
-	'data-firstday'        => $firstday,
-	'data-weekend'         => implode(',', $weekend),
+	'data-firstday'        => empty($firstday) ? '' : $firstday,
```
Comment (wilsonge) [inline layouts/joomla/form/field/calendar.php:106]: Is it going to be better to use hardcoded fallbacks here? Or are the fallbacks in the JS?

Comment (joomdonation) [reply layouts/joomla/form/field/calendar.php:106]: I don't know the details here, so @Fedik will give us the answer :).

Comment (Fedik) [reply layouts/joomla/form/field/calendar.php:106]: the JS have default value for these

Comment (Fedik) [reply layouts/joomla/form/field/calendar.php:106]: https://github.com/joomla/joomla-cms/blob/c9d977f8421cc5054eb54ecbdf8b4d6df8bf016a/build/media_source/system/js/fields/calendar.es5.js#L85-L87

and 

https://github.com/joomla/joomla-cms/blob/c9d977f8421cc5054eb54ecbdf8b4d6df8bf016a/build/media_source/system/js/fields/calendar.es5.js#L135-L138

Comment (wilsonge) [reply layouts/joomla/form/field/calendar.php:106]: Then wouldn't it be better to omit the attribute if it doesn't exist?

Comment (Fedik) [reply layouts/joomla/form/field/calendar.php:106]: It doesn't matter much, it will work both way (for these specific attributes).

Comment (Fedik) [reply layouts/joomla/form/field/calendar.php:106]: to me, the code looks more clear with empty one ;)

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on 52e1ac57d4c5f32b301e30f6497d68229f79a38f<br /><br />Error confirmed and patch works!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34949">issues.joomla.org/tracker/joomla-cms/34949</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34949">issues.joomla.org/tracker/joomla-cms/34949</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**